pyvista/core/filters/data_set.py

```diff
@@ -5213,3 +5218,128 @@ def integrate_data(self, progress_bar=False):
         alg.SetDivideAllCellDataByVolume(False)
         _update_alg(alg, progress_bar, 'Integrating Variables')
         return _get_output(alg)
+
+    def partition(self, n_partitions, generate_global_id=False, as_composite=True):
+        """Break down input dataset into requested number of partitions.
+
+        Cells on boundaries are uniquely to each partition.
+
+        It uses a DIY-based kdtree implementation that builds balances the cell
+        centers among requested number of partitions. Current implementation
+        only supports power-of-2 target partition. If a non-power of two value
+        is specified for ``n_partitions``, then the load balancing simply
+        uses the power-of-two greater than the requested value
+
+        For more details, see `vtkRedistributeDataSetFilter
+        <https://vtk.org/doc/nightly/html/classvtkRedistributeDataSetFilter.html>`_.
+
+        Parameters
+        ----------
+        n_partitions : int
+            Specify the number of partitions to split the input dataset
+            into. Current implementation results in number of partitions equal
+            to the power of 2 greater than or equal to the chosen value.
+
+        generate_global_id : bool, default: False
```

Comment:
This seems to be unused, and I'm unsure exactly what it does.

Reply:
Fixed in 8bcb10af7417fc5ab491b6922f193f7d3fd64ae5.